x/oracle/keeper/oracle.go

```diff
@@ -40,20 +40,27 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 			sdk.NewAttribute(types.AttributeKeyOracleAddress, oracleRegistration.OracleAddress),
 		),
 	)
+
 	return nil
 }
 
 func (k Keeper) ApproveOracleRegistration(ctx sdk.Context, msg *types.MsgApproveOracleRegistration) error {
-
-	if err := k.validateApproveOracleRegistration(ctx, msg); err != nil {
+	// validate approval for oracle registration
+	if err := k.validateApprovalSharingOracleKey(ctx, msg.GetApprovalSharingOracleKey(), msg.GetSignature()); err != nil {
 		return sdkerrors.Wrapf(types.ErrApproveOracleRegistration, err.Error())
 	}
 
+	// get oracle registration
 	oracleRegistration, err := k.GetOracleRegistration(ctx, msg.ApprovalSharingOracleKey.UniqueId, msg.ApprovalSharingOracleKey.TargetOracleAddress)
 	if err != nil {
 		return sdkerrors.Wrapf(types.ErrApproveOracleRegistration, err.Error())
 	}
 
+	// if EncryptedOraclePrivKey is already set, return error
+	if oracleRegistration.EncryptedOraclePrivKey != nil {
+		return sdkerrors.Wrapf(types.ErrApproveOracleRegistration, "already approved oracle registration. if you want to be shared oracle private key again, please register oracle again")
+	}
+
```

Comment:
Instead of checking that if oracle is set by `k.HasOracle()` which was in `validateApproveOracleRegistration`, it returns error if the `oracleRegistration.EncryptedOraclePrivKey` is not nil.
That means someone approved the oracle registration, so that the `EncryptedOraclePrivKey` is not nil.